### PR TITLE
Migrate CSV upload page

### PR DESCRIPTION
**Pull request by aghimpu**

### CODE REVIEW

File: react/src/dashboard/components/pages/csv_validations/elements/CSVUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+                const csvData = (results.data = results.data.filter((item) => !!item));
+                const header = csvData.length > 0 ? csvData[0] : [];
+
+                const body = (csvData.length > 0 ? csvData.slice(1) : []).filter((row) => {
+                    return Array.isArray(row) && row.filter((item) => !!item).length > 0;
+                });
+
+                const fundRecordKey =
+                    fund.csv_required_keys.filter((key) => {
+                        return key.indexOf('_eligible') === key.length - 9;
```
[the remark this answers is not in the document]
Comment (lexlog) [reply react/src/dashboard/components/pages/csv_validations/elements/CSVUpload.tsx:139]: @aghimpu could this be actual?

**Comment by lexlog** [comment]
note - third to test

**Review by irinaBerendeeva87 — COMMENTED** [review]
### QA review:

❌ During the loading of the transaction page, the Upload Transactions Batch Button appeared after other elements.
this is will be fixed in another task(https://github.com/teamforus/forus-frontend-react/pull/76)

https://github.com/teamforus/forus-frontend-react/assets/55141843/7e691bca-8f84-48c8-8dee-f9f85943f2f0

❌ Not appeared the button "Make a bulk transaction now" after the CSV file was uploaded. 

❌ When deleting a selected file before uploading, the Upload button is displayed.
Screenshot 2024-05-16 at 16 13 40

✅ notifications when uploading the correct/ incorrect file
✅ downloading sample file 
✅ screen with successful uploading
✅ uploading via drag-and-drop of a file 
✅ uploading file from a computer
✅ error screen when "Transaction creation failed!"

**Review by irinaBerendeeva87 — CHANGES_REQUESTED** [review]
**Checked:**

**Activatiecode aanmaken**
❌ field in forms : 
not all fields are present  in Stadjerspas form: 
Screenshot 2024-05-24 at 13 51 21

❌ input field validation: 
allow to Create activation code when BSN has value 1. 
Screenshot 2024-05-24 at 14 27 04

❌ Can't proceed with the form when filling in all visible fields. 
Screenshot 2024-05-24 at 14 05 24

✅ error messages 


✅ message when "No applicant"
Screenshot 2024-05-24 at 13 54 10

**Drag & drop functionality**
❌ Drag & drop  functionality causes white screen and errors in the console
Screenshot 2024-05-24 at 14 48 57

https://github.com/teamforus/forus-frontend-react/assets/55141843/35dfd50f-743e-420a-9df9-4d1b4cf7672e

✅ Paggination

**Comment by dev-rminds** [comment]
@irinaBerendeeva87 the last commit has react.sponsor branch from backend as dependency
image
https://github.com/teamforus/forus-backend/pull/2135
I should fix some issues migrated from angular version like missing label/placeholder for primary key

image
image

and record type key instead of the name in the table bellow:
image
image

Not sure how to reproduce:
- no error message when a sponsor with limited rights tries to delete Applicant details

For "allow to Create activation code when BSN has value 1" since it's not react related, we need to create a separate issue.

And also not react related, I think the part where you can add extra fields is not working properly:
image

It is possible to add more records but since the api will only allow to add records which are part of the criteria, formula or 
is primary_key (which are already in the form by default and cant be removed), anything you will add as extra will not pass the validation which makes the whole feature unusable.
I think this also needs a separate issue to investigate what are the use cases and how to properly fix it.

**Review by irinaBerendeeva87 — CHANGES_REQUESTED** [review]
**Retested:**
❌ modal "Activatiecode aanmaken" don't have bsn_hash  in description
Screenshot 2024-06-04 at 15 49 31
✅ all fields are present in Stadjerspas form
✅  Drag & drop functionality

**Review by irinaBerendeeva87 — DISMISSED** [review]
### Checked:
✅ modal "Activatiecode aanmaken" have bsn_hash in description
Screenshot 2024-06-07 at 13 07 24

**Comment by lexlog** [comment]
according to error https://github.com/teamforus/forus-frontend-react/pull/100#pullrequestreview-2104290552
@dev-rminds proposed to make a fallback for missing translation

**Review by irinaBerendeeva87 — APPROVED** [review]

**Pull request merged**